data/schema/game_config.cfg

```
@@ -80,17 +80,54 @@
             [/type]
         [/union]
     [/type]
-    # definition of range_list and s_range_list, before macros are expanded
-    {LIST_TYPE_COMPLEX range (
+    # definition of positive_range_list and s_positive_range_list, before macros are expanded
+    {LIST_TYPE_COMPLEX positive_range (
         [element]
             value="\d+"
         [/element]
         [element]
             value="\d+-\d+"
         [/element]
+        [element]
+            value="\d+\.\.\d+"
+        [/element]
         [element]
             value="\d+-infinity"
         [/element]
+        [element]
+            value="\d+\.\.infinity"
+        [/element]
+    )}
+    # definition of signed_range_list and s_signed_range_list, before macros are expanded
+    {LIST_TYPE_COMPLEX signed_range (
+        [element]
+            value="(-?)\d+"
+        [/element]
+        [element]
+            value="-\d+-(-?)\d+"
+        [/element]
+        [element]
+            value="-\d+\.\.(-?)\d+"
+        [/element]
+        [element]
+            value="-infinity-(-?)\d+"
+        [/element]
+        [element]
+            value="-infinity\.\.(-?)\d+"
+        [/element]
+        # If the first number is positive, the second one has to be too
+        [element]
+            value="\d+-\d+"
+        [/element]
+        [element]
+            value="\d+\.\.\d+"
+        [/element]
+        [element]
+            value="\d+-infinity"
+        [/element]
+        [element]
+            value="\d+\.\.infinity"
+        [/element]
```

This could actually just reference `positive_range` directly, since it's identical. That does mean that there are technically two elements that can potentially match a range like "9" but that doesn't seem like much of an issue.
```suggestion
        # If the first number is positive, the second one has to be too
        [element]
            link="positive_range"
        [/element]
```

Also, I believe you said it accepts `-infinity..infinity`, but you're rejecting it at the schema level? Is that intended or an oversight? I could see either way, honestly.

---

I feel that making a filter with "accept anything" is mistake, but one that isn't worth special casing in either location.